opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndexScan.java

```diff
@@ -80,7 +83,7 @@ public void open() {
     request = requestBuilder.build();
     iterator = Collections.emptyIterator();
     queryCount = 0;
-    fetchNextBatch();
+    rawResponse = fetchNextBatch().getRawResponse();
```

Comment:
There's a problem here.
 `fetchNextBatch` is called in `hasNext` when it's piecing together a scroll request. Scroll request is issued when limit clause specifies number larger than index window size. 
 
 Raw responses for each batch need to be added together to get raw response for the whole query.

Here's an integration test that demonstrates the problem. It gets count of rows in a table, runs the same SQL query in JDBC and JSON and confirms that count is the expected value (1000), JDBC response 1000 entries, and that JSON and JDBC responses have the same number of entries. That last assert fails -- JSON response only has 3 entries -- the  window value for this test.

```java
  @Test
  public void limitLargerThanWindow() throws IOException {
    setMaxResultWindow(TestsConstants.TEST_INDEX_ACCOUNT, 3);

    var count = executeJdbcRequest("SELECT COUNT(*) FROM " + TestsConstants.TEST_INDEX_ACCOUNT);
    var query = "SELECT * FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " LIMIT 1000";
    var requestBody = String.format("{\"query\" : \" %s\"}", query);
    var r = executeJdbcRequest(query);

    // The call below throws an exception
    var response = query(requestBody);
    var hits = response.getHits();
    assertEquals(1000, count.getJSONArray("datarows").getJSONArray(0).getInt(0));
    assertEquals(1000, r.getJSONArray("datarows").length());
    assertEquals(r.getJSONArray("datarows").length(), hits.length);
  }
```

Reply:
Legacy does not support having `max_result_window` less than limit size and throws the exception:
```
"error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "Result window is too large, from + size must be less than or equal to: [3] but was [1000]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting."
            }
        ]
```

In order to have parity with legacy, the plugin will also fall back to legacy when `LIMIT` is included in the query.